drivers/ina3221/ina3221.c

```diff
@@ -0,0 +1,928 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ina3221
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for Texas Instruments INA3221
+ *              three-channel, high-side current and bus voltage
+ *              monitor
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <periph/gpio.h>
+#include <byteorder.h>
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+#include "include/ina3221_defines.h"
+#include "include/ina3221_params.h"
+#include "include/ina3221_regs.h"
+#include "ina3221.h"
+
+struct channel_info
+{
+    uint8_t  chi_reg_shunt;
+    uint8_t  chi_reg_bus;
+    uint8_t  chi_reg_crit_alert_limit;
+    uint8_t  chi_reg_warn_alert_limit;
+};
+
+static struct channel_info _chi[INA3221_NUM_CH] = {
+    (struct channel_info){
+        .chi_reg_shunt            = INA3221_REG_CH1_SHUNT_VOLTAGE,
+        .chi_reg_bus              = INA3221_REG_CH1_BUS_VOLTAGE,
+        .chi_reg_crit_alert_limit = INA3221_REG_CH1_CRIT_ALERT_LIMIT,
+        .chi_reg_warn_alert_limit = INA3221_REG_CH1_WARN_ALERT_LIMIT
+    },
+    (struct channel_info){
+        .chi_reg_shunt            = INA3221_REG_CH2_SHUNT_VOLTAGE,
+        .chi_reg_bus              = INA3221_REG_CH2_BUS_VOLTAGE,
+        .chi_reg_crit_alert_limit = INA3221_REG_CH2_CRIT_ALERT_LIMIT,
+        .chi_reg_warn_alert_limit = INA3221_REG_CH2_WARN_ALERT_LIMIT
+    },
+    (struct channel_info){
+        .chi_reg_shunt            = INA3221_REG_CH3_SHUNT_VOLTAGE,
+        .chi_reg_bus              = INA3221_REG_CH3_BUS_VOLTAGE,
+        .chi_reg_crit_alert_limit = INA3221_REG_CH3_CRIT_ALERT_LIMIT,
+        .chi_reg_warn_alert_limit = INA3221_REG_CH3_WARN_ALERT_LIMIT
+    },
+};
+
+/**
+ * @brief Convert register value to shunt voltage in uV
+ * 
+ * @param[in]   reg_val Register value
+ * 
+ * @pre         reg_val must be in host byte order
+ * 
+ * @returm      Shunt voltage in uV
+ */
+static inline int32_t reg_val_to_shunt_voltage_uV(int16_t reg_val)
+{
+    //assert(reg_val >= INA3221_MIN_SHUNT_REG_VAL);
```

Comment:
Why commented out?

Reply:
Compiler reported it will always evaluate true. I will remove it.